modyn/tests/trainer_server/internal/trainer/remote_downsamplers/test_abstract_matrix_downsampling_strategy.py

```diff
@@ -34,30 +34,30 @@ def test_init():
 @patch.multiple(AbstractMatrixDownsamplingStrategy, __abstractmethods__=set())
 def test_collect_embeddings():
     amds = AbstractMatrixDownsamplingStrategy(*get_sampler_config())
+    with torch.inference_mode(mode=(not amds.requires_grad)):
+        amds.matrix_content = MatrixContent.EMBEDDINGS
```

Comment:
I think simply moving this line to above the `with` statement would work, i.e. so that `requires_grad()` does not need to consider None. Right?

Reply:
probably, but i actually prefer the idea of moving this to the constructor to avoid those none fields!